lib/module.js

```diff
@@ -3,47 +3,31 @@ const { promisify } = require('util')
 const { removeSync, outputFile } = require('fs-extra')
 const { Feed } = require('feed')
 const AsyncCache = require('async-cache')
-const consola = require('consola')
-
-const logger = consola.withScope('nuxt:feed')
+const logger = require('./logger')
 
 const defaults = {
   path: '/feed.xml',
   async create(feed) {},
   cacheTime: 1000 * 60 * 15
 }
 
-async function feedModule() {
-  // Factory function
-  if (typeof this.options.feed === 'function') {
-    this.options.feed = await this.options.feed()
-  }
-
-  // Factory object
-  if (!Array.isArray(this.options.feed)) {
-    if (this.options.feed.factory) {
-      this.options.feed = await this.options.feed.factory(this.options.feed.data)
-    }
-  }
-
-  // Single feed
-  if (!Array.isArray(this.options.feed)) {
-    this.options.feed = [this.options.feed]
-  }
-
-  const options = Object.assign([], this.options.feed).map(o => Object.assign({}, defaults, o))
+async function feedModule(moduleOptions) {
+  const options = [
+    ...await parseOptions(moduleOptions),
+    ...await parseOptions(this.options.feed)
+  ].map(o => ({ ...{}, ...defaults, ...o }))
 
   const feedCache = new AsyncCache({
     load(feedIndex, callback) {
-      createFeed(options[feedIndex], callback).catch(err => /* istanbul ignore next */ logger.error(err))
+      createFeed(options[feedIndex], callback).catch(err => logger.error(err))
     }
   })
 
   feedCache.get = promisify(feedCache.get)
 
   options.forEach((feedOptions, index) => {
-    this.nuxt.hook('generate:before', async () => {
-      const xmlGeneratePath = resolve(this.options.srcDir, join('static', feedOptions.path))
+    this.nuxt.hook('generate:done', async () => {
```

Comment:
This is a breaking change

Reply:
What do you recommend?

Reply:
A separate PR for this change :wink:

Reply:
@manniL rollback, after approving this PR I will create a new one with this change